pkg/api/v20170701/types.go

```diff
@@ -5,6 +5,20 @@ import (
 	"strings"
 )
 
+// Add the validate tag to add a mark, if this is a required or optional field
+// Current it is not used in any logic yet
+// TODO, we could possibly use this tag in all the Validate functions
+const (
+	ValidateTag = "validate"
```

Comment:
Instead of calling it "validate" , why not "required" or "isrequired"?
E.g.:
Location string     `json:"location,omitempty" isrequired:"true" 

or

Location string     `json:"location,omitempty" required:"true"